ledger/internal/eval.go

```diff
@@ -918,26 +913,26 @@ func (eval *BlockEvaluator) TestTransaction(txn transactions.SignedTxn) error {
 // Transaction tentatively adds a new transaction as part of this block evaluation.
 // If the transaction cannot be added to the block without violating some constraints,
 // an error is returned and the block evaluator state is unchanged.
-func (eval *BlockEvaluator) Transaction(txn transactions.SignedTxn, ad transactions.ApplyData) error {
+func (eval *BlockEvaluator) Transaction(txn transactions.SignedTxn, ad transactions.ApplyData, spVCgetter ledgercore.StateProofTrackerGetter) error {
 	return eval.TransactionGroup([]transactions.SignedTxnWithAD{
 		{
 			SignedTxn: txn,
 			ApplyData: ad,
 		},
-	})
+	}, spVCgetter)
 }
 
 // TransactionGroup tentatively adds a new transaction group as part of this block evaluation.
 // If the transaction group cannot be added to the block without violating some constraints,
 // an error is returned and the block evaluator state is unchanged.
-func (eval *BlockEvaluator) TransactionGroup(txads []transactions.SignedTxnWithAD) error {
-	return eval.transactionGroup(txads)
+func (eval *BlockEvaluator) TransactionGroup(txads []transactions.SignedTxnWithAD, spVCgetter ledgercore.StateProofTrackerGetter) error {
```

Comment:
IMO it does not make sense polluting Transaction / TransactionGroup args list. How about setting the SP stuff at BlockEvaluator construction time?

Reply:
I agree with you 100%
I just wanted to keep it here so that we can easily see any loose ends, and one is in the indexer code, where nil is passed. 
If you agree that this approach is better than the interfaces, then that can be easily done.